backend/src/main/java/org/eventplanner/events/services/ImoListService.java

```diff
@@ -0,0 +1,175 @@
+                if (cell != null && cell.getCellType() != CellType.BLANK) {
+                    break;
+                } else if (cell != null && cell.getCellType() == CellType.STRING && cell.getStringCellValue().trim().isEmpty()) {
+                    break;
+                } else if (!cellIterator.hasNext()) {
+                    rowIsEmpty = true;
+                }
+            }
+        }
+        return rowCounter;
+    }
+
+    private void writeCrewDataToSheet(XSSFSheet sheet, List<Registration> crewList, Map<PositionKey, Position> positionMap) throws IOException{
+
+        int firstEmptyRow = findFirstEmptyRow(sheet);
+
+        for (int crewCounter = 1; crewCounter < crewList.size(); crewCounter++){
+            Row currentRow = sheet.getRow(crewCounter+firstEmptyRow-1);
+            currentRow.getCell(0).setCellValue(crewCounter);
+            Registration currentRegistration = crewList.get(crewCounter);
+            String imoListRank = ObjectUtils.mapNullable(positionMap.get(currentRegistration.getPosition()),
+                    Position::getImoListRank,
+                    currentRegistration.getPosition().value());
+            UserKey crewMemberKey = currentRegistration.getUser();
+            if (crewMemberKey != null) {
+                Optional<UserDetails> crewMemberDetails = userService.getUserByKey(crewMemberKey);
+
+                if (crewMemberDetails.isPresent()) {
+                    currentRow.getCell(1).setCellValue(crewMemberDetails.get().getLastName());
+                    currentRow.getCell(2).setCellValue(crewMemberDetails.get().getFirstName());
+                    currentRow.getCell(3).setCellValue(imoListRank);
+                    currentRow.getCell(4).setCellValue(crewMemberDetails.get().getNationality());
+                    var dateOfBirth = crewMemberDetails.get().getDateOfBirth();
+                    if (dateOfBirth != null) {
+                        currentRow.getCell(5).setCellValue(dateOfBirth.format(DateTimeFormatter.ofPattern("dd.MM.yyyy")));
+                    } else {
+                        currentRow.getCell(5).setCellValue("");
+                    }
+                    currentRow.getCell(6).setCellValue(crewMemberDetails.get().getPlaceOfBirth());
+                    currentRow.getCell(7).setCellValue(crewMemberDetails.get().getPassNr());
+                    continue;
+                }
+            }
+            // user not found
+            currentRow.getCell(1).setCellValue(currentRegistration.getName());
+            currentRow.getCell(2).setCellValue("");
+            currentRow.getCell(3).setCellValue(imoListRank);
+            currentRow.getCell(4).setCellValue("");
+            currentRow.getCell(5).setCellValue("");
+            currentRow.getCell(6).setCellValue("");
+            currentRow.getCell(7).setCellValue("");
+        }
+    }
+
+    private void addEventDetailsToImoList(XSSFSheet sheet,String placeHolder, String eventDetail){
```

Comment:
This functions name should be more generic, like "replacePlaceholderInSheet" or something like that, because it is not limited to events
space after ,